subcommand/webhook-cert-manager/command.go

```diff
@@ -0,0 +1,277 @@
+			return 1
+		}
+	}
+
+	configFile, err := ioutil.ReadFile(c.flagConfigFile)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("error reading config file from %s", c.flagConfigFile))
+	}
+	var configs []webhookConfig
+	err = json.Unmarshal(configFile, &configs)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("error unmarshalling config file: %s", err.Error()))
+	}
+
+	// Create the certificate notifier so we can update for certificates,
+	// then start all the background routines for updating certificates.
+	ctx := context.Background()
+	ctx, cancelFunc := context.WithCancel(ctx)
+	defer cancelFunc()
+	certCh := make(chan cert.MetaBundle)
+
+	var notifiers []*cert.Notify
+
+	for _, config := range configs {
+		certSource := &cert.GenSource{
+			Name:  "Consul Webhook Certificates",
+			Hosts: config.TLSAutoHosts,
+		}
+		certNotify := &cert.Notify{Source: certSource, Ch: certCh, WebhookName: config.Name, SecretName: config.SecretName, SecretNamespace: config.SecretNamespace}
+		notifiers = append(notifiers, certNotify)
+		go certNotify.Start(ctx)
+	}
+
+	go c.certWatcher(ctx, certCh, c.clientset, c.UI)
+
+	closeNotifiers := func() {
+		for _, notifier := range notifiers {
+			notifier.Stop()
+		}
+	}
+	defer closeNotifiers()
+
+	// Interrupted, gracefully exit
+	select {
+	case <-c.sigCh:
+		closeNotifiers()
+		cancelFunc()
+		return 0
+	}
+}
+
+func (c *Command) certWatcher(ctx context.Context, ch <-chan cert.MetaBundle, clientset kubernetes.Interface, log cli.Ui) {
+	var bundle cert.MetaBundle
+	for {
+		select {
+		case bundle = <-ch:
+			log.Info("updated certificate bundle received. updating webhook certs.")
+			// Bundle is updated, set it up
+
+		case <-time.After(30 * time.Minute):
+			// This forces the mutating ctrlWebhook config to remain updated
+			// fairly quickly. This is done every 30 minutes to ensure the certificates
+			// are in sync. Because the certificate and key are being read from a secret,
+			// this does not have to be processed as aggressively as the 1 sec time in
+			// the connect inject cert watcher.
+
+		case <-ctx.Done():
+			// Quit
+			return
+		}
+
+		log.Info("getting secret from kubernetes")
+		certSecret, err := clientset.CoreV1().Secrets(bundle.SecretNamespace).Get(ctx, bundle.SecretName, metav1.GetOptions{})
+		if err != nil && k8serrors.IsNotFound(err) {
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      bundle.SecretName,
+					Namespace: bundle.SecretNamespace,
+				},
+				Data: map[string][]byte{
+					corev1.TLSCertKey:       bundle.Cert,
+					corev1.TLSPrivateKeyKey: bundle.Key,
+				},
+				Type: corev1.SecretTypeTLS,
+			}
+
+			log.Info("creating kubernetes secret")
+			if _, err := clientset.CoreV1().Secrets(bundle.SecretNamespace).Create(ctx, secret, metav1.CreateOptions{}); err != nil {
+				log.Error(fmt.Sprintf("error writing secret to API server: %s", err))
+				continue
+			}
+
+			log.Info("updating webhook configuration with new CA")
```

Comment:
I think this CA will only be new when cert manager starts, and after that it'll keep updating it with the same CA.
```suggestion
			log.Info("updating webhook configuration")
```